### DISCUSSION TITLE
Use APIM in combination with DAB - 'nextLink URL'

### DISCUSSION THREAD

**Discussion by dennismeijer84** [Azure/data-api-builder#1447] [Q&A] [answered]
We're exposing all our API's using Azure API Management and are also looking into exposing DAB using APIM (currently hosted on ACI).   At this moment we can succesfully call the DAB endpoint through APIM, but the 'nextLink' url contains the hostname of the Container instance. 

We can search/replace the url (`http://xxxx.westeurope.azurecontainer.io:5000/api` with the custom domain of APIM) in an APIM policy, but it doesn't feel like the optimal solution. Is there any solution available to 'override' the baseUrl and path in the response body with 'nextLink' results so we can expose the DAB through APIM (or only get a relative path?) ?

**Comment by JerryNixon** [comment]
I think APIM support is important. Here are a couple of options:

1. Could do now: [APIM can rewrite the payload and replace the nextLink substring with the APIM path](https://learn.microsoft.com/en-us/azure/api-management/set-body-policy). 
2. Could do later: In the dab.config add `nextLinkRelative: true` or something like it.

Right now, at least, the first option is a plausible workaround. No?

What I am not clear on here is if returning a relative link is sufficient for all APIM support.

**Reply by dennismeijer84** [reply, marked as answer]
Indeed, the APIM policy was also the solution I was thinking about as a workaround. I've added the following code to the outbound section of the APIM policy (quick and dirty setup):

`<find-and-replace from="@(context.Request.Url.Scheme + "://" + context.Request.Url.Host + ":" + context.Request.Url.Port + context.Request.Url.Path)" to="@(context.Request.OriginalUrl.Scheme + "://" + context.Request.OriginalUrl.Host + context.Request.OriginalUrl.Path)" />`

**Reply by seantleonard** [reply]
Thank you for sharing your workaround, @dennismeijer84!
